src/services/quandl-service.js

```diff
@@ -113,16 +163,42 @@
             });
         }
 
-        getData(stockCode, cb) {
+        /**
+        * @param stockCode {String} Stock code to query
+        * @param cb {Function} callback to be called on success and error
+        * @todo use alternative API key instead of defaulting to No key
+        * @todo should we show a warning to the user when we swap to anonymous?
+        */
+        getData(stockCode, cb, isRetry = false) {
             return this.stockData().get({ code: stockCode }, (result) => {
-                var stock = {
-                    name: result.dataset.name,
+                if (!useApiKey && !isRetry) {
+                    this._notify('CONNECTION_STATUS_CHAGED', (useApiKey = true));
```

Comment:
Oh yeah, the typo. Took me 30 seconds of starring at it to realize what you mean :)